### DISCUSSION TITLE
Can't I use wire:click in a non-component view?

### DISCUSSION THREAD

**Discussion by duran004** [livewire/livewire#8882] [1. Help]
I am making a simple Cart system with livewire. I am trying to dispatch the method from an unrelated view file with button wire:click. But it doesn't work. If I do it from Cart Livewire Component, it works. 
 how do I make this work? 
```
<button wire:click=“console.log(‘Button Clicked’)”>Add to Cart</button>
```
This is how I check if the event is triggered
```
<script>
            document.addEventListener('livewire:init', () => {
                console.log('Livewire started');
                Livewire.on('addToCart', (data) => {
                    console.log('Event triggered:', data);
                });
            });
        </script>
```
 the event is triggered via the console. 
```
Livewire.dispatch('addToCart', { productId: 1, productName: 'qweqwe', price: 100 })
```
This is not working:
```
<button wire:click="Livewire.dispatch('addToCart', { productId: 1, productName: 'qweqwe', price: 100 })">Add to Cart</button>
```
For this example I can simply replace wire:click with onclick, but I may want to use other properties starting with wire:

referenced here: https://livewire.laravel.com/docs/events#dispatching-events-from-blade-templates

**Comment by lrljoe** [comment]
wire:click is only valid within livewire components

Check the docs for how to do this within JS/Alpine
